src/components/ui/select.tsx

```diff
@@ -0,0 +1,176 @@
+import * as React from "react";
+import * as SelectPrimitive from "@radix-ui/react-select";
+import { Check, ChevronDown, ChevronUp } from "lucide-react";
+
+import { cn } from "@/lib/utils";
+
+const Select = SelectPrimitive.Root;
+
+const SelectGroup = SelectPrimitive.Group;
+
+const SelectValue = SelectPrimitive.Value;
+
+const SelectTrigger = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Trigger>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Trigger> & {
+    label?: string;
+  }
+>(({ className, children, label, ...props }, ref) => {
+  return (
+    <div className="relative w-full">
+      {label && (
+        <label
```

Comment:
can we change the colour of the label depending on whether its selected or not per the figma design.

You might find the peer modifier useful
https://tailwindcss.com/docs/hover-focus-and-other-states